book/developers/exex/tracking-state.md

```diff
@@ -0,0 +1,193 @@
+# Tracking State
+
+In this chapter, we'll learn how to keep track of some state inside our ExEx.
+
+Let's continue with our Hello World example from the [previous chapter](./hello-world.md).
+
+### Turning ExEx into a struct
+
+First, we need to turn our ExEx into a stateful struct.
+
+Before, we had just an async function, but now we'll need to implement
+the [`Future`](https://doc.rust-lang.org/std/future/trait.Future.html) trait manually.
+
+<div class="warning">
+
+Having a stateful async function is also possible, but it makes testing harder,
+because you can't access variables inside the function to assert the state of your ExEx.
+
+</div>
+
+```rust,norun,noplayground,ignore
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{ready, Context, Poll},
+};
+
+use reth::api::FullNodeComponents;
+use reth_exex::{ExExContext, ExExEvent, ExExNotification};
+use reth_node_ethereum::EthereumNode;
+use reth_tracing::tracing::info;
+
+struct MyExEx<Node: FullNodeComponents> {
+    ctx: ExExContext<Node>,
+}
+
+impl<Node: FullNodeComponents> Future for MyExEx<Node> {
+    type Output = eyre::Result<()>;
+
+    fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
```

Comment:
We should also explain how you'd call out to other functions?

Prob we'd save most of the logic in async functions on the exex that the future loop calls?